### PR TITLE
[buteo-sync-plugins-social] Support downsyncing Google Calendar reminders. Contributes to MER#1071

### PR DESCRIPTION
**Pull request by chriadam**
This commit adds support for downsyncing Google Calendar reminders.
It does not add support for upsyncing reminder data to Google.

Contributes to MER#1071

### CODE REVIEW

**Comment by chriadam** [comment]
@pvuorela @pgerdt @antseppa

File: src/google/google-calendars/googlecalendarsyncadaptor.cpp [hunk 1]
```diff
@@ -565,6 +565,60 @@ void jsonToKCal(const QJsonObject &json, KCalCore::Event::Ptr event, KCalCore::I
...
+                SOCIALD_LOG_DEBUG("setting event reminder with interval (seconds):" << interval);
+                for (int i = 0; i < alarms.count(); ++i) {
+                    if (alarms.at(i)->type() == KCalCore::Alarm::Procedure) {
+                        continue;
+                    }
+                    event->removeAlarm(alarms.at(i));
+                }
+                KCalCore::Alarm::Ptr alarm = event->newAlarm();
+                alarm->setEnabled(true);
+                alarm->setStartOffset(KCalCore::Duration(interval * -1));
```
Comment (pvuorela) [inline src/google/google-calendars/googlecalendarsyncadaptor.cpp]: What if alarm is removed remotely after being synced?

Comment (chriadam) [inline src/google/google-calendars/googlecalendarsyncadaptor.cpp]: Good point, will add support for that, thanks.

**Comment by chriadam** [comment]
Updated to support deleting alarms, and also now it converts the Google StartOffset into a "valid" Nemo StartOffset (as defined by https://github.com/nemomobile/nemo-qml-plugin-calendar/blob/master/src/calendarworker.cpp#L326 to display in the UI).

Note that this means that we cannot upsync alarms without causing "spurious" server-side changes, as the local data will be different to the server data, by design.

I personally don't think that this is a big problem.  Certainly, until we decide to support upsync of alarms, it's not a problem at all.

File: src/google/google-calendars/googlecalendarsyncadaptor.cpp [hunk 2]
```diff
@@ -539,7 +539,102 @@ void extractRecurrence(const QJsonArray &recurrence, KCalCore::Event::Ptr event,
     }
 }
 
-void jsonToKCal(const QJsonObject &json, KCalCore::Event::Ptr event, KCalCore::ICalFormat &icalFormat)
+int nearestNemoReminderStartOffset(int googleStartOffset)
+{
+    // Google supports arbitrary start offsets, whereas in Nemo's UI
+    // we only allow specific reminder offsets.
+    // See nemo-qml-plugin-calendar::NemoCalendarEvent::Reminder for
+    // those offset definitions.
```
Comment (pvuorela) [inline src/google/google-calendars/googlecalendarsyncadaptor.cpp]: Nasty dependency, we should really at least allow showing custom reminder times in the calendar, but I guess we can start with rounding first.

**Comment by pvuorela** [comment]
LGTM

**Pull request merged**